feat(ourlogs): Add logs to replays

k-fish opened the pull request:
### Summary
Adds logs to the replays tab, querying via traceid first then fetching all logs.

Closes LOGS-86

k-fish commented on static/app/views/replays/detail/ourlogs/index.tsx:
```diff
@@ -0,0 +1,129 @@
+  return (
+    <SearchQueryBuilderProvider {...searchQueryBuilderProps}>
+      <PaddedFluidHeight ref={scrollContainerRef} fullHeight={!hasInfiniteFeature}>
+        {hasInfiniteFeature ? (
+          <LogsInfiniteTable
+            stringAttributes={stringAttributes}
+            numberAttributes={numberAttributes}
+            showHeader
+            allowPagination
+            scrollContainer={scrollContainerRef}
+          />
+        ) : (
+          <LogsTable showHeader={false} allowPagination />
+        )}
```
This should be infinite table only soon

billyvg commented on static/app/views/replays/detail/layout/focusTabs.tsx:
```diff
@@ -32,9 +36,10 @@ type Props = {
 function FocusTabs({isVideoReplay}: Props) {
   const organization = useOrganization();
   const {getActiveTab, setActiveTab} = useActiveReplayTab({isVideoReplay});
+  const hasLogsFeature = useOrganization().features.includes('ourlogs-enabled');
```
```suggestion
  const hasLogsFeature = organization.features.includes('ourlogs-enabled');
```

billyvg replied:
what is our rollout status for this flag?

billyvg reviewed (commented):
<img width="715" alt="image" src="https://github.com/user-attachments/assets/fd83e3cc-f098-4846-953c-7459cd4d8df1" />

missing borders

--

Also there's two diff loading states:

- "Loading logs..." text inside of a panel (i.e. with borders)
- loading spinner, no panel
- switching into the Logs tab always shows "Loading logs"
- pagination is always loading indicator

billyvg commented on static/app/views/replays/detail/layout/focusTabs.tsx:
```diff
@@ -32,9 +36,12 @@ type Props = {
   const {getActiveTab, setActiveTab} = useActiveReplayTab({isVideoReplay});
+  const hasLogsFeature =
+    organization.features.includes('ourlogs-enabled') &&
+    organization.features.includes('ourlogs-infinite-scroll');
```
I don't see a ff for this, could you add the replay team so we can poke around?

k-fish replied:
I've been using our overrides more recently rather than options automator, do you not see it?
![Screenshot 2025-06-02 at 2 19 01 PM](https://github.com/user-attachments/assets/eea25021-b277-4469-a22b-6672391f8c6b)

billyvg reviewed (approved):
There's some design "quirks" in here, but I think this is fine to slam in if we're still iterating and this isn't available for users yet.

k-fish commented:
@billyvg can you list them so I can track them? I'll fix them before we turn on infinite scroll

billyvg commented:
@k-fish 

* Needs some margin here 
![image](https://github.com/user-attachments/assets/971221c2-7565-49d6-8488-fcc2b7356525)
* Headers are a bit too big: 
![image](https://github.com/user-attachments/assets/d7bc2eef-4dd4-421f-a6ba-b60393c25efd)
they should match our other tables:
![image](https://github.com/user-attachments/assets/bceb297b-6e2b-4e23-b93e-dfb31661c670)
* Ideally scrolling should be inside of the panel body (see Network table)
* The panel should fill the tab area (height-wise): ![image](https://github.com/user-attachments/assets/955a6458-ced9-4413-9d82-889c17447fe9)

k-fish committed:
feat(ourlogs): Add logs to replays tab

This refactors the original code committed to use the new infinite table, and makes sure to work closer to how other replay details tab works (local search etc.). Currently this is behind a feature flag  so the tab will not currently appear for our users.

k-fish committed:
Add Open in Logs button to replays detail page

k-fish committed:
Fix scroll container

k-fish committed:
knip

k-fish committed:
Get rid of extra if

k-fish committed:
Fix styles

k-fish commented on static/app/views/explore/logs/useLogsSearchQueryBuilderProps.tsx:
```diff
@@ -0,0 +1,71 @@
+
+      setLogsPageParams({
+        search: newSearch,
+        fields: newColumns.length ? [...fields, ...newColumns] : undefined,
```
That was what I was thinking, but I'm realizing now this behaviour might be non-standard, I think having the columns added helps narrow down the value when doing wildcards specifically

k-fish committed:
Merge remote-tracking branch 'origin/master' into feat/ourlogs-add-to-replay-tab
